test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -584,10 +584,14 @@ func BenchmarkPerfScheduling(b *testing.B) {
 		b.Run(tc.Name, func(b *testing.B) {
 			for _, w := range tc.Workloads {
 				b.Run(w.Name, func(b *testing.B) {
+					// 30 minutes should be plenty enough even for the 5000-node tests.
+					ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Minute)
```

Comment:
TODO has the connotation that we still have something to fix.

```suggestion
					ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
```

Reply:
... which in this case is true: `context.TODO` should be replaced with a context that has a logger for per-test output. This was done in a separate commit of this PR, but then got taken out because it is not strictly related to it.

I'm fine with keeping the `context.TODO` and will follow up shortly after this PR is merged.

Reply:
ah, sounds good.